conans/tools.py

```diff
@@ -165,6 +165,17 @@ def build_sln_command(settings, sln_path, targets=None, upgrade_project=True, bu
         command += " /target:%s" % ";".join(targets)
     return command
 
+def vs_installation_path(version):
+    if not hasattr(vs_installation_path, "_cached"):
```

Comment:
Basically yes, Capturing all, not knowing what will be raised hide unexpected errors in your code, maybe something is not programmed well or some software piece changes the behaviour. 
It's not so bad:
- The `os.environ` can return a default value, if None return.
- wswhere.exe and any other error can return a != 0 code or an exception I don't exactly know how works the subprocess.
- About the decode, yes, capture it. May be interesting to log the error to improve the decode.